server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -597,6 +584,17 @@ public void doRun()
           }
         }
     );
+    handoffNotifier.registerSegmentHandoffCallback(
```

Comment:
changes in this file include - 
- replacing serverView with handoffNotifier 
- it registers callback with handoffNotifier when pushing a segment
- moved the logic for determining when handoff is complete from plumber to the handoff notifier.